server/repository/src/migrations/views/mod.rs

```diff
@@ -156,16 +156,28 @@ pub(crate) fn rebuild_views(connection: &StorageConnection) -> anyhow::Result<()
         invoice.status AS invoice_status,
         invoice_line_stock_movement.total_before_tax AS total_before_tax,
         invoice_line_stock_movement.pack_size as pack_size,
-        invoice_line_stock_movement.number_of_packs as number_of_packs
+        invoice_line_stock_movement.number_of_packs as number_of_packs,
+        CASE
+          WHEN invoice.type IN ('INBOUND_SHIPMENT', 'CUSTOMER_RETURN', 'INVENTORY_ADDITION') THEN 1
+          WHEN invoice.type IN ('OUTBOUND_SHIPMENT', 'SUPPLIER_RETURN', 'PRESCRIPTION', 'INVENTORY_REDUCTION') THEN 2
+          ELSE 3
+        END AS type_precedence
     FROM
         invoice_line_stock_movement
         LEFT JOIN reason_option ON invoice_line_stock_movement.reason_option_id = reason_option.id
         LEFT JOIN stock_line ON stock_line.id = invoice_line_stock_movement.stock_line_id
         JOIN invoice ON invoice.id = invoice_line_stock_movement.invoice_id
         JOIN name_link ON invoice.name_link_id = name_link.id
         JOIN name ON name_link.name_id = name.id
+    ORDER BY datetime, type_precedence
     )
-    SELECT * FROM all_movements
+    SELECT *,
+      SUM(quantity) OVER (
+        PARTITION BY store_id, stock_line_id
+        ORDER BY datetime, type_precedence
+        ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW
+      ) AS running_balance
+     FROM all_movements
```

Comment:
We don't want to add it here because we're using this view for other things too~

Reply:
What things? We could just not use the field where it doesn't make sense?

Reply:
Thanks, have updated 🙏  [6799050](https://github.com/msupply-foundation/open-msupply/pull/8366/commits/6799050ff98fb3cdde23245859bd0af31467fe9e)

Also added balance column to the stock ledger tab seeing as we have it now 👀